stdlib/public/core/AutoDiff.swift

```diff
@@ -766,6 +766,57 @@ public struct AnyDerivative : Differentiable & AdditiveArithmetic {
   }
 }
 
+/// A pair of type-erased derivatives.
+public struct AnyDerivativePair : Differentiable & AdditiveArithmetic {
+  public var first: AnyDerivative
+  public var second: AnyDerivative
+
+  @differentiable(vjp: _vjpInit(_:_:))
+  public init(_ first: AnyDerivative, _ second: AnyDerivative) {
+    self.first = first
+    self.second = second
+  }
+
+  @usableFromInline internal static func _vjpInit(
```

Comment:
Manual implementations of `_vjpInit`, `_vjpAdd`, and `_vjpSubtract` should not be necessary.
It should be possible to simply mark `init`, `+` and `-` with `@differentiable`.

However, this is blocked by [TF-429](https://bugs.swift.org/browse/TF-429). The stdlib is compiled with `-enable-resilience`, causing primal value structs to be resilient.